packages/go_router/lib/src/route.dart

```diff
@@ -458,4 +558,771 @@ class ShellRoute extends RouteBase {
+/// StatefulShellRoute could be achieved. In this example, a
+/// BottomNavigationBar with two tabs is used, and each of the tabs gets its
+/// own Navigator. A container widget responsible for managing the Navigators
+/// for all route branches will then be passed as the child argument
+/// of the builder function.
+///
+/// ```
+/// final GlobalKey<NavigatorState> _tabANavigatorKey =
+///   GlobalKey<NavigatorState>(debugLabel: 'tabANavigator');
+/// final GlobalKey<NavigatorState> _tabBNavigatorKey =
+///   GlobalKey<NavigatorState>(debugLabel: 'tabBNavigator');
+///
+/// final GoRouter _router = GoRouter(
+///   initialLocation: '/a',
+///   routes: <RouteBase>[
+///     StatefulShellRoute(
+///       builder: (BuildContext context, GoRouterState state,
+///             StatefulNavigationShell navigationShell) {
+///         return ScaffoldWithNavBar(navigationShell: navigationShell);
+///       },
+///       navigatorContainerBuilder: (BuildContext context,
+///             StatefulNavigationShell navigationShell,
+///             List<Widget> children) =>
+///                 MyCustomContainer(children: children),
+///       branches: [
+///         /// The first branch, i.e. tab 'A'
+///         StatefulShellBranch(
+///           navigatorKey: _tabANavigatorKey,
+///           routes: <RouteBase>[
+///             GoRoute(
+///               path: '/a',
+///               builder: (BuildContext context, GoRouterState state) =>
+///                   const RootScreen(label: 'A', detailsPath: '/a/details'),
+///               routes: <RouteBase>[
+///                 /// Will cover screen A but not the bottom navigation bar
+///                 GoRoute(
+///                   path: 'details',
+///                   builder: (BuildContext context, GoRouterState state) =>
+///                       const DetailsScreen(label: 'A'),
+///                 ),
+///               ],
+///             ),
+///           ],
+///         ),
+///         /// The second branch, i.e. tab 'B'
+///         StatefulShellBranch(
+///           navigatorKey: _tabBNavigatorKey,
+///           routes: <RouteBase>[
+///             GoRoute(
+///               path: '/b',
+///               builder: (BuildContext context, GoRouterState state) =>
+///                   const RootScreen(label: 'B', detailsPath: '/b/details'),
+///               routes: <RouteBase>[
+///                 /// Will cover screen B but not the bottom navigation bar
+///                 GoRoute(
+///                   path: 'details',
+///                   builder: (BuildContext context, GoRouterState state) =>
+///                       const DetailsScreen(label: 'B'),
+///                 ),
+///               ],
+///             ),
+///           ],
+///         ),
+///       ],
+///     ),
+///   ],
+/// );
+/// ```
+///
+/// See [Stateful Nested Navigation](https://github.com/flutter/packages/blob/main/packages/go_router/example/lib/stacked_shell_route.dart)
```

Comment:
Use breadcumb like this one https://github.com/flutter/flutter/blob/5df903a0134a2c062abca9007ea9fde8b8e65d2a/packages/flutter/lib/src/widgets/router.dart#L526

also mention the indexedStack name constructor